### PR TITLE
Add `notice tips` override pdf tpls in module exemple

### PR DESCRIPTION
**Pull request by PululuK**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Add `notice tips` override pdf tpls in module exemple
| Fixed ticket? | N/A

### CODE REVIEW

File: src/content/1.7/modules/concepts/pdf.md [hunk 1]
```diff
@@ -291,3 +291,168 @@ public function hookActionGetExtraMailTemplateVars($hookArgs)
     $hookArgs['extra_template_vars']['{password}'] = '*******'; 
 }
 ```
+
+
+### Code exemple : 
+
+{{% notice tip %}}
+You can override the core pdf tpls in your module.
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:299]:
```suggestion
You can override the core PDF templates in your module.
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 2]
```diff
@@ -291,3 +291,168 @@ public function hookActionGetExtraMailTemplateVars($hookArgs)
...
+* PHP Code 
+
+```php
+   
+<?php
+if (!defined('_PS_VERSION_')) {
+    exit;
+}
+
+class Mycustompdfgen extends Module
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:330]:
```suggestion
class MyCustomPdfGenerator extends Module
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 3]
```diff
@@ -291,3 +291,168 @@ public function hookActionGetExtraMailTemplateVars($hookArgs)
...
+{
+    public function __construct()
+    {
+        $this->name = 'mycustompdfgen';
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:334]:
```suggestion
        $this->name = 'mycustompdfgenerator';
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 4]
```diff
@@ -291,3 +291,168 @@ public function hookActionGetExtraMailTemplateVars($hookArgs)
...
+    }
+    
+    public function install()
+    {
+        return parent::install();
+    }
+    public function uninstall()
+    {
+       return parent::uninstall();
+    }
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:357]:
```suggestion
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 5]
```diff
@@ -291,3 +291,168 @@ public function hookActionGetExtraMailTemplateVars($hookArgs)
...
+   
+    public function genPDF(array $params)
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:359]:
```suggestion
    public function generatePDF(array $params): void
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 6]
```diff
@@ -291,3 +291,168 @@ public function hookActionGetExtraMailTemplateVars($hookArgs)
...
+    {
+        $myOrderObject = new Order((int) $params['id_order']);
+       
+        $myCustomSlipVarsForPdfContent = $this->myContentDatasPresenter($myOrderObject);
+        $myCustomSlipVarsForPdfFooter  = $this->myFooterDatasPresenter($myOrderObject);
+        $myCustomSlipVarsForPdfHeader  = $this->myHeaderDatasPresenter($myOrderObject);
+
+
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:367]:
```suggestion

```

File: src/content/1.7/modules/concepts/pdf.md [hunk 7]
```diff
@@ -291,3 +291,168 @@ public function hookActionGetExtraMailTemplateVars($hookArgs)
...
+       
+        $pdfGen = new PDFGenerator(false, 'P');
+		$pdfGen->setFontForLang(Context::getContext()->language->iso_code);
+		$pdfGen->startPageGroup();
+		$pdfGen->createHeader($this->getHeader($myCustomSlipVarsForPdfHeader));
+		$pdfGen->createFooter($this->getFooter($myCustomSlipVarsForPdfHeader));
+		$pdfGen->createContent($this->getPdfContent($myCustomSlipVarsForPdfHeader));
+		$pdfGen->writePage();
+		$pdfGen->render('my_custom_pdf.pdf', 'D');
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:375]:
```suggestion
        $pdfGen = new PDFGenerator(false, 'P');
	$pdfGen->setFontForLang(Context::getContext()->language->iso_code);
	$pdfGen->startPageGroup();
	$pdfGen->createHeader($this->getHeader($myCustomSlipVarsForPdfHeader));
	$pdfGen->createFooter($this->getFooter($myCustomSlipVarsForPdfHeader));
	$pdfGen->createContent($this->getPdfContent($myCustomSlipVarsForPdfHeader));
	$pdfGen->writePage();
	$pdfGen->render('my_custom_pdf.pdf', 'D');
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 8]
```diff
@@ -291,3 +291,168 @@ public function hookActionGetExtraMailTemplateVars($hookArgs)
...
+   }
+   
+   /**
+     * Returns the template's HTML content.
+     *
+     * @return string HTML content
+     */
+    public function getPdfContent(array $myCustomSlipVarsForPdfContent)
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:383]:
```suggestion
    public function getPdfContent(array $myCustomSlipVarsForPdfContent): string
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 9]
```diff
@@ -291,3 +291,168 @@ public function hookActionGetExtraMailTemplateVars($hookArgs)
...
+
+        return $this->context->smarty->fetch(__DIR__.'/pdf/delivery-slip.tpl');
+    }
+
+   /**
+     * Returns the template's HTML footer.
+     *
+     * @return string HTML footer
+     */
+    public function getFooter(array $myCustomSlipVarsForPdfFooter)
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:402]:
```suggestion
    public function getFooter(array $myCustomSlipVarsForPdfFooter): string
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 10]
```diff
@@ -291,3 +291,168 @@ public function hookActionGetExtraMailTemplateVars($hookArgs)
...
+        $this->context->smarty->assign($myCustomSlipVarsForPdfFooter);
+        return $this->context->smarty->fetch(__DIR__.'/pdf/footer.tpl');
+    }
+
+    /**
+     * Returns the template's HTML header.
+     *
+     * @return string HTML header
+     */
+    public function getHeader(array $myCustomSlipVarsForPdfHeader)
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:413]:
```suggestion
    public function getHeader(array $myCustomSlipVarsForPdfHeader): string
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 11]
```diff
@@ -291,3 +291,168 @@ public function hookActionGetExtraMailTemplateVars($hookArgs)
...
+    }
+   
+   
+   /**
+   * Format your order data here for pdf content : ['tpl_var_name'=>'tpl_value']
+   *
+   * @return array
+   */
+   public function myContentDatasPresenter(Order $myOrderObject): array{
+     // TODO : impletent
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:426]:
```suggestion
     // TODO : implement it 
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 12]
```diff
@@ -291,3 +291,168 @@ public function hookActionGetExtraMailTemplateVars($hookArgs)
...
+        return $this->context->smarty->fetch(__DIR__.'/pdf/header.tpl');
+    }
+   
+   
+   /**
+   * Format your order data here for pdf content : ['tpl_var_name'=>'tpl_value']
+   *
+   * @return array
+   */
+   public function myContentDatasPresenter(Order $myOrderObject): array{
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:425]:
```suggestion
   public function myContentDatasPresenter(Order $myOrderObject): array
   {
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 13]
```diff
@@ -291,3 +291,168 @@ public function hookActionGetExtraMailTemplateVars($hookArgs)
...
+     // TODO : impletent 
+   }
+   
+   /**
+   * Format your order data here for pdf footer : ['tpl_var_name'=>'tpl_value']
+   *
+   * @return array
+   */
+   public function myFooterDatasPresenter(Order $myOrderObject): array {
+     // TODO : impletent
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:435]:
```suggestion
   public function myFooterDatasPresenter(Order $myOrderObject): array
   {
     // TODO : implement it 
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 14]
```diff
@@ -291,3 +291,168 @@ public function hookActionGetExtraMailTemplateVars($hookArgs)
...
+     // TODO : impletent 
+   }
+   
+   /**
+   * Format your order data here for pdf header : ['tpl_var_name'=>'tpl_value']
+   *
+   * @return array
+   */
+   public function myHeaderDatasPresenter(Order $myOrderObject): array{
+     // TODO : impletent
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:444]:
```suggestion
   public function myHeaderDatasPresenter(Order $myOrderObject): array
   {
     // TODO : implement it 
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 15]
```diff
@@ -291,3 +291,168 @@ public function hookActionGetExtraMailTemplateVars($hookArgs)
...
+   public function myHeaderDatasPresenter(Order $myOrderObject): array{
+     // TODO : impletent 
+   }
+}
+```
+
+* Usage 
+
+```php
+ $myModuleOrderPdfGen = Module::getInstanceByName('mycustompdfgen');
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:452]:
```suggestion
 $myModuleOrderPdfGenerator = Module::getInstanceByName('mycustompdfgenerator');
```

File: src/content/1.7/modules/concepts/pdf.md [hunk 16]
```diff
@@ -291,3 +291,168 @@ public function hookActionGetExtraMailTemplateVars($hookArgs)
...
+ $myModuleOrderPdfGen->genPDF([
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/pdf.md:453]:
```suggestion
 $myModuleOrderPdfGenerator->generatePDF([
```

**Review by Progi1984 — COMMENTED** [review]
Some (a lot of) feedbacks

**Review by Progi1984 — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @PululuK

**Pull request merged**